From 8cca3569c695fc5886a7a4c2392323ea03636442 Mon Sep 17 00:00:00 2001
From: remittor <remittor@gmail.com>
Date: Mon, 7 Sep 2015 14:48:39 +0300
Subject: [PATCH] Add support property "persist.call_recording.enabled"

---
 .../android/services/callrecorder/CallRecorderService.java   | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/com/android/services/callrecorder/CallRecorderService.java b/src/com/android/services/callrecorder/CallRecorderService.java
index 10e7c7f..0ad44b9 100644
--- a/src/com/android/services/callrecorder/CallRecorderService.java
+++ b/src/com/android/services/callrecorder/CallRecorderService.java
@@ -50,6 +50,7 @@ public class CallRecorderService extends Service {
     private RecorderState mState = RecorderState.IDLE;
     private CallRecording mCurrentRecording = null;
 
+    private static final String ENABLE_PROPERTY = "persist.call_recording.enabled";
     private static final String AUDIO_SOURCE_PROPERTY = "persist.call_recording.src";
 
     private SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyMMdd_HHmmssSSS");
@@ -110,7 +111,7 @@ public class CallRecorderService extends Service {
         int formatValue =  Settings.System.getInt(
                 getContentResolver(), Settings.System.CALL_RECORDING_FORMAT, mDefaultEncoder);
         if (formatValue == 0){
-            return MediaRecorder.OutputFormat.AMR_WB;
+            return MediaRecorder.OutputFormat.AMR_NB;
         } else {
             return MediaRecorder.OutputFormat.MPEG_4;
         }
@@ -120,7 +121,7 @@ public class CallRecorderService extends Service {
         int formatValue =  Settings.System.getInt(
                 getContentResolver(), Settings.System.CALL_RECORDING_FORMAT, mDefaultEncoder);
         if (formatValue == 0){
-            return MediaRecorder.AudioEncoder.AMR_WB;
+            return MediaRecorder.AudioEncoder.AMR_NB;
         } else {
             return MediaRecorder.AudioEncoder.HE_AAC;
         }
@@ -208,14 +209,15 @@ public class CallRecorderService extends Service {
     private String generateFilename(String number) {
         String timestamp = DATE_FORMAT.format(new Date());
         int audioFormat = getAudioFormat();
-        if (audioFormat == MediaRecorder.OutputFormat.AMR_WB){
+        if (audioFormat == MediaRecorder.OutputFormat.AMR_NB){
             return number + "_" + timestamp + ".amr";
         } else {
-            return number + "_" + timestamp + ".m4a ";
+            return number + "_" + timestamp + ".m4a";
         }
     }
 
     public static boolean isEnabled(Context context) {
-        return context.getResources().getBoolean(R.bool.call_recording_enabled);
+        boolean defaultValue = context.getResources().getBoolean(R.bool.call_recording_enabled);
+        return SystemProperties.getBoolean(ENABLE_PROPERTY, defaultValue);
     }
 }
-- 
1.9.5.msysgit.0

