From 53d05a8cb31a87eb5a360647f5b23d7e1834b3d8 Mon Sep 17 00:00:00 2001
From: remittor <remittor@gmail.com>
Date: Mon, 18 Jan 2016 14:20:32 +0300
Subject: [PATCH] Add hook functions to post_process_props.py

---
 tools/post_process_props.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tools/post_process_props.py b/tools/post_process_props.py
index 64af01d..1daab0d 100755
--- a/tools/post_process_props.py
+++ b/tools/post_process_props.py
@@ -14,7 +14,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import sys
+import os, sys
+from post_process_props_hook import *
 
 
 def iteritems(obj):
@@ -136,8 +137,10 @@ def main(argv):
 
   if filename.endswith("/build.prop"):
     mangle_build_prop(properties, extraargs)
+    mangle_build_prop_hook(properties, extraargs)
   elif filename.endswith("/default.prop"):
     mangle_default_prop(properties)
+    mangle_default_prop_hook(properties)
   else:
     sys.stderr.write("bad command line: " + str(argv) + "\n")
     sys.exit(1)
-- 
1.9.5.msysgit.0

